static/js/components/UserListItems.js

```diff
@@ -0,0 +1,192 @@
+  LearningResourceCard,
+  LearningResourceRow
+} from "../components/LearningResourceCard"
+import { SEARCH_LIST_UI } from "../lib/search"
+import {
+  userListItemsRequest,
+  updateUserListItemPositionMutation,
+  createUserListItemsSelector,
+  createUserListItemsPageSelector
+} from "../lib/queries/user_list_items"
+
+import type { ResourceItem } from "../lib/queries/user_list_items"
+import type { UserList } from "../flow/discussionTypes"
+
+type UserListItemsProps = {
+  userListId: number,
+  items: Array<ResourceItem>,
+  className: string,
+  isSorting: ?boolean
+}
+
+type ItemsProps = {
+  userListId: number,
+  className: string,
+  items: Array<ResourceItem>
+}
+
+type UserListItemSortableCardsProps = {
+  userListId: number,
+  className: string,
+  isSorting: ?boolean,
+  pageSize: number
+}
+
+type PaginatedUserListItemsProps = {
+  userList: UserList,
+  pageSize: number
+}
+
+const SortItems = ({ userListId, items, className }: ItemsProps) => {
+  const [{ isPending }, updateUserListItemPosition] = useMutation(
+    updateUserListItemPositionMutation
+  )
+
+  return (
+    <SortableContainer
+      shouldCancelStart={() => isPending}
+      onSortEnd={async ({ oldIndex, newIndex }) => {
+        const item = items[oldIndex]
+        await updateUserListItemPosition(
+          userListId,
+          item.item.item_id,
+          oldIndex,
+          newIndex
+        )
+      }}
+    >
+      <Grid className={isPending ? `${className} saving` : className}>
+        {items.map((item, i) => (
+          <SortableItem key={`item-${item.item.item_id}`} index={i}>
+            <Cell width={12}>
+              <LearningResourceCard
+                object={item.resource}
+                searchResultLayout={SEARCH_LIST_UI}
+                reordering
+              />
+            </Cell>
+          </SortableItem>
+        ))}
+      </Grid>
+    </SortableContainer>
+  )
+}
+
+const ViewItems = ({ items, className }: ItemsProps) => {
+  return (
+    <Grid className={className}>
+      {items.map(item => (
+        <Cell width={12} key={`item-${item.item.item_id}`}>
+          <LearningResourceCard
+            object={item.resource}
+            searchResultLayout={SEARCH_LIST_UI}
+          />
+        </Cell>
+      ))}
+    </Grid>
+  )
+}
+
+const SortAndViewItems = ({ isSorting, ...props }: UserListItemsProps) => {
+  return isSorting ? <SortItems {...props} /> : <ViewItems {...props} />
+}
+
+const loader = <Loading className="infinite" key="loader" />
+
+export function UserListItemSortableCards({
+  userListId,
+  className,
+  isSorting,
+  pageSize
+}: UserListItemSortableCardsProps) {
+  const [page, setPage] = useState(0)
+
+  const [{ isFinished }] = useRequest(
+    userListItemsRequest(userListId, pageSize * page, pageSize)
+  )
+  const initialLoadDone = page !== 0 || isFinished
+
+  const { items, count } = useSelector(createUserListItemsSelector(userListId))
+
+  return initialLoadDone ? (
+    <InfiniteScroll
```

Comment:
After changing the limit from 50 to 10 (to match the `max_limit` value of `DefaultPagination`), when I get to the bottom of the page, a request for the next 10 is made, but then cancelled as the next set is requested:

<img width="246" alt="Screen Shot 2020-01-09 at 10 56 21 AM" src="https://user-images.githubusercontent.com/187676/72083153-52621480-32cf-11ea-8afb-5f351ea2ce9e.png">

If I continue to try scrolling down, further requests are made far beyond the actual number of items in the list:

<img width="125" alt="Screen Shot 2020-01-09 at 10 56 51 AM" src="https://user-images.githubusercontent.com/187676/72083219-6c035c00-32cf-11ea-9a73-8f2ba9d9286b.png">